clang/runtime/dpct-rt/include/dpct/device.hpp

```diff
@@ -662,8 +663,16 @@ class dev_mgr {
   void select_device(unsigned int id) {
     std::lock_guard<std::recursive_mutex> lock(m_mutex);
     check_id(id);
-    _thread2dev_map[get_tid()] = id;
+    auto curr_tid = get_tid();
+
+    // make sure the device stack exist
```

Comment:
`select_device` shouldn't push current device into the stack

Reply:
Now we are only using a device stack for all CUDA context operation as discussed.